src/Maestro/Maestro.Data/BuildAssetRegistryContext.cs

```diff
@@ -77,6 +76,9 @@ public class BuildAssetRegistryContext : IdentityDbContext<ApplicationUser, Iden
         public DbSet<DependencyFlowEvent> DependencyFlowEvents { get; set; }
         public DbSet<GoalTime> GoalTime { get; set; }
         public DbSet<LongestBuildPath> LongestBuildPaths { get; set; }
+        // This is for testing purpose only. Instead of adding records to the db, this will let us add 
+        // record to the list.
+        public List<RepositoryBranchUpdateHistoryEntry> RepoBranchUpdateInMemory { get; set; }
```

Comment:
Is there a compelling reason to add this test property to the real class instead of creating a "TestBuildAssetRegistryContext"  class?

Reply:
I agree. The real code shouldn't be doing this.  This might mean something needs to be virtual, or some sort of inheritance, but the testing code shouldn't intrude into the real code.

Reply:
The issue is the DbQuery configuration. the OnModelCreating method needs to configure the query to use for each DbQuery property. For the real implementation the DbQuery uses a raw SQL statement. The In-Memory provider can't handle that SQL, so we have to change how the query is set up. We might be able to just override the OnModelCreating in a new class and set up the query again with the testing implementation. I don't know if the model builder thingy can handle setting up the same query twice.

Reply:
The only thing that even uses this property is the "SaveAsync" method.  It should just call some virtual method that get's overridden by the mock.

Reply:
Sorry, being tricked by GitHub.  Whatever the giant method is that's using it... that code that is green here can just be moved into a virtual method that's overridden by the mock.

Reply:
The thing that needs to be changed for the tests is the RepositoryBranchUpdateHistory property, I don't think EF would handle just overriding that property implementation to not be a normal property correctly.

Reply:
It's a new property we just added.  Add it to the Mock and not to the real class.

Reply:
The new property is there only so the `builder.Query<RepositoryBranchUpdateHistoryEntry>()` call in OnModelCreating can use it. That is how the DbQuery properties are set up.

Reply:
If you are telling me that EF isn't testable, I guess that's fine, just one more thing to pile on my giant stack of reasons I dislike EF, but that seems unlikely to be true.  What would happen if this property were put in a sub class, and the code in green was made virtual?  Would it somehow just cause the huge pile of magic that is EF to burst into flame?

Reply:
It might work if we put things in a sub class. I am not sure though.

Reply:
I will try this and update this thread.

Reply:
I was able to extract this to a separate class. :)

Reply:
I added this as a separate class, so resolving this conversation.